src/main/scala/com/microsoft/hyperspace/index/sources/iceberg/IcebergRelation.scala

```diff
@@ -123,14 +125,26 @@ class IcebergRelation(spark: SparkSession, override val plan: DataSourceV2Relati
    *
    * File paths should be the same format as "input_file_name()" of the given relation type.
    * For [[IcebergRelation]], each file path should be in this format:
-   *   `/path/to/file`
+   *   `/path/to/file` or `X:/path/to/file`
    *
    * @param fileIdTracker [[FileIdTracker]] to create the list of (file path, file id).
    * @return List of pairs of (file path, file id).
    */
   override def lineagePairs(fileIdTracker: FileIdTracker): Seq[(String, Long)] = {
-    fileIdTracker.getFileToIdMap.toSeq.map { kv =>
-      (kv._1._1.replaceAll("^file:/{1,3}", "/"), kv._2)
+    // For Windows,
+    //   original file path: file:/C:/path/to/file
+    //   input_file_name(): C:/path/to/file
+    // For Linux,
+    //   original file path: file:///path/to/file or file:/path/to/file
+    //   input_file_name(): /path/to/file
+    if (System.getProperty("os.name").toLowerCase(Locale.ROOT).contains("windows")) {
+      fileIdTracker.getFileToIdMap.toSeq.map { kv =>
+        (kv._1._1.stripPrefix("file:/"), kv._2)
+      }
+    } else {
+      fileIdTracker.getFileToIdMap.toSeq.map { kv =>
+        (kv._1._1.replaceAll("^file:/{1,3}", "/"), kv._2)
+      }
```

Comment:
```suggestion
    val isWindowsOS =
      System.getProperty("os.name").toLowerCase(Locale.ROOT).contains("windows")
    val replacement = if (isWindowsOS) { "" } else { "/" }
    fileIdTracker.getFileToIdMap.toSeq.map { kv =>
      (kv._1._1.replaceAll("^file:/{1,3}", replacement), kv._2)
```

Reply:
I prefer the current one; I think it looks clearer for os specific impl.
Thanks!

Reply:
I understand your preference... and in the same time I would contradict you saying " it looks clearer for os specific impl" - I don't think there can be anything clearer than `if (isWindowsOS)` construct that I proposed :).

Reply:
either approach seems fine to me.